src/components/job/JobResultModal.tsx

```diff
@@ -0,0 +1,197 @@
+  Paper,
+} from '@mui/material';
+import { Close } from '@mui/icons-material';
+import { Action } from 'redux';
+import { connect } from 'react-redux';
+import ReactJson from 'react-json-view';
+import { Property } from 'csstype';
+import { getJobResult } from 'actions';
+import { TdrState } from 'reducers';
+import { JobResult, JobResultError } from 'reducers/job';
+import LoadingSpinner from '../common/LoadingSpinner';
+
+const styles = (theme: CustomTheme) => ({
+  root: {
+    boxShadow: 'none',
+    maxHeight: '100%',
+    position: 'relative' as Property.Position,
+  },
+  seeMoreLink: {
+    cursor: 'pointer',
+    border: 'none',
+    backgroundColor: 'transparent',
+    color: theme.palette.primary.main,
+    '&:hover': {
+      color: theme.palette.primary.hover,
+    },
+  },
+  dialog: {
+    minHeight: '80vh',
+    maxHeight: '80vh',
+    width: '80%',
+    maxWidth: 800,
+  },
+  dialogTitle: {
+    margin: 0,
+    marginTop: '5px',
+    fontSize: '1.2rem',
+  },
+  dialogInfo: {
+    display: 'flex',
+    marginBottom: 10,
+  },
+  dialogLabel: {
+    fontWeight: 'bold',
+    minWidth: 100,
+    'text-align': 'right',
+    marginRight: 15,
+  },
+  dialogContent: {
+    'word-break': 'break-all',
+  },
+});
+
+type JobResultModalProps = {
+  classes: ClassNameMap;
+  dispatch: Dispatch<Action>;
+  id: string;
+  description?: string;
+  jobClass?: string;
+  loading: boolean;
+  jobResult?: JobResult;
+  linkDisplay?: any;
+};
+
+function JobResultModal({
+  classes,
+  dispatch,
+  id,
+  description,
+  jobClass,
+  loading,
+  jobResult,
+  linkDisplay = 'See More',
+}: JobResultModalProps) {
+  const [seeMore, setSeeMore] = useState({ open: false });
+
+  useEffect(() => {
+    if (seeMore.open) {
+      dispatch(getJobResult({ id }));
+    }
+  }, [dispatch, seeMore, id]);
+
+  const handleSeeMoreOpen = () => {
+    setSeeMore({ open: true });
+  };
+
+  const handleSeeMoreClose = () => {
+    setSeeMore({ open: false });
+  };
+
+  const jobError =
+    jobResult && jobResult.resultType === 'error' ? (jobResult.result as JobResultError) : null;
+
+  const jobSuccess =
+    jobResult && jobResult.resultType === 'success' ? (jobResult.result as any) : null;
+
+  return (
+    <div>
+      <button type="button" onClick={handleSeeMoreOpen} className={classes.seeMoreLink}>
+        {linkDisplay}
+      </button>
+      <Paper className={classes.root}>
+        <Dialog
+          open={seeMore.open}
+          scroll="paper"
+          fullWidth={true}
+          classes={{ paper: classes.dialog }}
+          onBackdropClick={handleSeeMoreClose}
+        >
+          <DialogTitle id="see-more-dialog-title">
+            <div className={classes.dialogTitle} style={{ float: 'left' }}>
+              Job Details
+            </div>
+            <IconButton size="small" style={{ float: 'right' }} onClick={handleSeeMoreClose}>
+              <Close />
+            </IconButton>
+          </DialogTitle>
+          <DialogContent>
```

Comment:
It looks like we have a similar dialog component in [LightTable](https://github.com/DataBiosphere/jade-data-repo-ui/blob/develop/src/components/table/LightTable.tsx#L395). Would it make sense to pull the dialog container plus close control into a reusable component? I'm thinking that could help keep styling/functionality uniform across the UI, but it might be more work than worth it.

Reply:
I dont know! The advantage to styling / functionality would be great, but if we wanted to customize a title to have a warning icon / message, or a "settings" icon, it could faster to edit component.

I feel like the dialog component already contains a lot of the similar styling, just coming from material ui itself already - The only classes that I think would be shared are `root` and `dialogTitle`, and I can drop them to make it just use the default instead?

Reply:
True, keeping it separate sounds like it makes sense then. Thanks for entertaining the idea!